mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -736,15 +762,27 @@ class GeneralizedLinearRegressionModel private[ml] (
   private lazy val familyAndLink = new FamilyAndLink(familyObj, linkObj)
 
   override protected def predict(features: Vector): Double = {
-    val eta = predictLink(features)
+    if (!isSet(offsetCol) || $(offsetCol).isEmpty) {
+      val eta = BLAS.dot(features, coefficients) + intercept
+      familyAndLink.fitted(eta)
+    } else {
+      throw new SparkException("Must supply offset value when offset is set.")
```

Comment:
maybe a better error message might be:
Must supply offset to predict when offset column is specified